src/index.js

```diff
@@ -1,390 +1,586 @@
+            element.setAttributeNS(XLINK_NS, name, nextValue)
+          }
+        } else {
+          if (nullOrFalse) {
+            element.removeAttribute(name)
+          } else {
+            element.setAttribute(name, nextValue)
+          }
+        }
+      }
     }
   }
+}
 
-  function resolveNode(node) {
-    return typeof node === "function"
-      ? resolveNode(node(globalState, wiredActions))
-      : node != null
-        ? node
-        : ""
+var createElement = function(node, lifecycle, eventProxy, isSvg) {
+  var element =
+    node.type === TEXT_NODE
+      ? document.createTextNode(node.name)
+      : (isSvg = isSvg || node.name === "svg")
+        ? document.createElementNS(SVG_NS, node.name)
+        : document.createElement(node.name)
+
+  var props = node.props
+  if (props.onCreate) {
+    lifecycle.push(function() {
+      props.onCreate(element)
+    })
   }
 
-  function render() {
-    skipRender = !skipRender
+  for (var i = 0, length = node.children.length; i < length; i++) {
+    element.appendChild(
+      createElement(node.children[i], lifecycle, eventProxy, isSvg)
+    )
+  }
 
-    var node = resolveNode(view)
+  for (var name in props) {
+    updateProperty(element, name, null, props[name], eventProxy, isSvg)
+  }
+
+  return (node.element = element)
+}
 
-    if (container && !skipRender) {
-      rootElement = patch(container, rootElement, oldNode, (oldNode = node))
+var updateElement = function(
+  element,
+  lastProps,
+  nextProps,
+  lifecycle,
+  eventProxy,
+  isSvg,
+  isRecycled
+) {
+  for (var name in merge(lastProps, nextProps)) {
+    if (
+      (name === "value" || name === "checked"
+        ? element[name]
+        : lastProps[name]) !== nextProps[name]
+    ) {
+      updateProperty(
+        element,
+        name,
+        lastProps[name],
+        nextProps[name],
+        eventProxy,
+        isSvg
+      )
     }
+  }
 
-    isRecycling = false
+  var cb = isRecycled ? nextProps.onCreate : nextProps.onUpdate
+  if (cb != null) {
+    lifecycle.push(function() {
+      cb(element, lastProps)
+    })
+  }
+}
 
-    while (lifecycle.length) lifecycle.pop()()
+var removeChildren = function(node) {
+  for (var i = 0, length = node.children.length; i < length; i++) {
+    removeChildren(node.children[i])
   }
 
-  function scheduleRender() {
-    if (!skipRender) {
-      skipRender = true
-      setTimeout(render)
-    }
+  var cb = node.props.onDestroy
+  if (cb != null) {
+    cb(node.element)
   }
 
-  function clone(target, source) {
-    var out = {}
+  return node.element
+}
 
-    for (var i in target) out[i] = target[i]
-    for (var i in source) out[i] = source[i]
+var removeElement = function(parent, node) {
+  var remove = function() {
+    parent.removeChild(removeChildren(node))
+  }
 
-    return out
+  var cb = node.props && node.props.onRemove
+  if (cb != null) {
+    cb(node.element, remove)
+  } else {
+    remove()
   }
+}
+
+var getKey = function(node) {
```

Comment:
This method will not be needed if we ensure that node.key is by default null. This can be done using default parameter in functions related to node.